kitsune/wiki/views.py

```diff
@@ -1445,3 +1473,116 @@ def what_links_here(request, document_slug):
     }
 
     return render(request, 'wiki/what_links_here.html', c)
+
+
+def get_fallback_locale(doc, request):
+    """Get best fallback local based on locale Mapping"""
+
+    # Get the Translated locales which have current revision
+    translated_locales = (doc.translations.all()
+                                          .values_list('locale', flat=True)
+                                          .exclude(current_revision=None))
+    # Get the Accepted_language header from user request
+    browser_language_code = request.META.get('HTTP_ACCEPT_LANGUAGE', None)
+    if browser_language_code is not None:
+        header_locales = parse_accept_lang_header(browser_language_code)
+    else:
+        header_locales = []
+    # Get the Locale mapping from setting
+    NON_SUPPORTED_LOCALES = settings.NON_SUPPORTED_LOCALES
+
+    # Check if the document is translated to any of the locale listed in Accepted_Language Header
+    for locale in header_locales:
+        if locale in translated_locales:
+            fallback_locale = locale
+            break
+    else:
+        # If any locale of Accpeted_Language Header is not in translated locale,
```

Comment:
Entire section rewrite:
# If there are no versions matching the locales listed in the ACCEPT_LANGUAGE header, check for fallback locales defined for locales listed in the ACCEPT_LANGUAGE header
# If the document is available in a fallback locale, show that locale
# The fallback locales are defined in the FALLBACK_LOCALES array in kitsune/wiki/config.py